best-practices/README.md

```diff
@@ -13,6 +13,12 @@ General
 * [Keep the code simple](http://www.readability.com/~/ko2aqda2).
 * Avoid premature optimizations. Get a version 1.0 out there as soon as you
   can. Until you have some users to measure, you're optimizing based on guesses.
+* Use `bin/setup` for initial setup of the project and for development
+  environment seed data.
+* Use `bin/server` for starting all the processes required to run the project
+  in a development environment, when suitable.
+* Use `bin/deploy` for deploying the project to staging and production
```

Comment:
Right, for static sites yes, we can probably keep it.

I also use parity, just not for deploys. What can you do with that you
can't from CI, regarding deploys?
On Wed, 17 Aug 2016 at 05:32, Miguel Palhas <notifications@github.com>
wrote:

> In best-practices/README.md
> <https://github.com/subvisual/guides/pull/21#discussion_r75091164>:
>
> > @@ -13,6 +13,12 @@ General
> >  * [Keep the code simple](http://www.readability.com/~/ko2aqda2).
> >  * Avoid premature optimizations. Get a version 1.0 out there as soon as you
> >    can. Until you have some users to measure, you're optimizing based on guesses.
> > +* Use `bin/setup` for initial setup of the project and for development
> > +  environment seed data.
> > +* Use `bin/server` for starting all the processes required to run the project
> > +  in a development environment, when suitable.
> > +* Use `bin/deploy` for deploying the project to staging and production
>
> TBH, you haven't personally been working on several conferences / landing
> pages, which is the reason you haven't used bin/deploy much.
>
> That said, I do agree that this rule is a bit meaningless nowadays, and we
> don't really need it.
> The only projects that have a bin/deploy are the ones hosted on gh-pages
> (and maybe some older apps).
> And given that there are so many variables (does the project have
> staging/production? do I need a password, and if so, where do I get it?
> what If I want to deploy a different branch?), this isn't as
> straightforward as a setup or a server to easily dump into a script
>
> Also, for Heroku apps, I use parity
>
> I would propose something like this instead:
>
>
>    - Prefer CI deploys over local ones, and provide instructions on how
>    they work (are they manually triggered or automatic? Is there a staging /
>    review apps set up?)
>    - If CI deploys aren't an option, provide clear instructions on how to
>    handle them in your README
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/subvisual/guides/pull/21/files/f868161eb4bfe9df631ec5f63701f4f796a858db#r75091164>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKJoM5pnm6Tve9lmBylr-Q2jd0fJgsCks5qgtVEgaJpZM4JlrFE>
> .
>